src/scvi/dataloaders/_data_splitting.py

```diff
@@ -34,27 +38,77 @@ def validate_data_split(
         Size of train set. Need to be: 0 < train_size <= 1.
     validation_size
         Size of validation set. Need to be 0 <= validation_size < 1
+    use_external_indexing
+        Whether to use external supported indexing. This bypass any other flag or input parameter
+        that was before
+    external_indexing
+        A list of np.arrays that is always in the order of [[train_idx],[valid_idx],[test_idx]].
     """
-    if train_size > 1.0 or train_size <= 0.0:
-        raise ValueError("Invalid train_size. Must be: 0 < train_size <= 1")
+    if use_external_indexing:
+        if external_indexing is None:
+            raise ValueError("External indexing list is missing")
+
+        if type(external_indexing) is not list:
+            raise ValueError("External indexing is not of list type")
+
+        # validate the structure of it
+        if type(external_indexing[0]) is not np.ndarray:
+            raise ValueError("The first element of the list of lists is not a np.array")
+
+        if external_indexing[1] is None:
+            # we can assume the 2nd element is not exists or empty list
+            # (i.e, only train index are supported)
+            external_indexing[1] = np.array([])  # empty np.array
+        if external_indexing[2] is None:
+            # we can assume the 3rd element is not exists or empty list (
+            # i.e, only train and valid index are supported)
+            external_indexing[2] = np.array([])  # empty np.array
+        if type(external_indexing[1]) is not np.ndarray:
+            raise ValueError("The second element of the list of lists is not a np.array")
+        if type(external_indexing[2]) is not np.ndarray:
+            raise ValueError("The third element of the list of lists is not a np.array")
+
+        # check for total number of indexes (overlapping or missing)
+        if (
+            len(external_indexing[0]) + len(external_indexing[1]) + len(external_indexing[2])
+        ) > n_samples:
+            raise ValueError("There are overlapping indexing please fix")
```

Comment:
An edge case for this check is if any of the indexes contains duplicate elements, e.g., the train split having `[1, 2, 3, 1]`.

Reply:
One way to address this would be to convert each element in the input list to a numpy set. This will also make the `np.intersect1d` checks below faster.

Reply:
I would keep only the intersection part and add np.unique(s)

I didn't think about duplicates.
The problem is not only here, the user can insert duplicate indexes per set (not overlapping) and its train/test/valid will hold duplicate rows. I don't think it makes sense. but maybe he will want to up/down sample for some reason.

I added to raise a warning for duplicates per set